typed-racket-test/unit-tests/typecheck-tests.rkt

```diff
@@ -122,6 +125,38 @@
     (define golden (golden-fun expanded-expr))
     (check-tc-results result golden #:name "tc-expr"))
 
+  ;; test-type-table : syntax? (listof (cons/c identifier? tc-results?)) -> void?
+  ;; Typecheck the given expression
+  ;;  and assert that the type-table has an entry for every identifier
+  ;;  in the given association list.
+  ;;
+  ;; Example: `(test-type-table expr (list (cons f t)))` will
+  ;; 1. type-check `expr`
+  ;; 2. search `expr` for occurrences of `f`
+  ;; 3. check that the type-table entry for `f` matches `t`
+  (define (test-type-table expr assoc)
```

Comment:
@samth @pnwamk any comments about this new testing function?

The goal is to make sure the `type-table` has the right mappings after TR type-checks a syntax object.